backend-lib/src/main/rust/lib.rs

```diff
@@ -797,16 +727,14 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_ge
         let addr = utils::java_string_to_rust(&env, address);
         let taddr = TransparentAddress::decode(&network, &addr).unwrap();
 
-        // We select all transparent funds including unmined coins, as that is the same
-        // set of UTXOs we shield from.
-        let min_confirmations = 0;
+        let min_confirmations = NonZeroU32::new(1).unwrap();
 
         let amount = (&db_data)
             .get_target_and_anchor_heights(min_confirmations)
             .map_err(|e| format_err!("Error while fetching anchor height: {}", e))
             .and_then(|opt_anchor| {
                 opt_anchor
-                    .map(|(_, a)| a)
+                    .map(|(target, _)| target) // Include unconfirmed funds.
```

Comment:
I continue to believe that showing the user balances that are completely unconfirmed is a bad idea. As I said [here](https://github.com/zcash/librustzcash/pull/914#discussion_r1314177526):
> If I send you funds in a bunch of transactions that each have 50 unpaid actions and expire at the next block, I can be pretty confident that at most one (probably zero) of them will be mined. So, using this value as defined for anything opens the wallet up to the transparent equivalent of a Faerie Gold attack (and to various social engineering attacks, such as refund scams, if it is ever shown to a user).